.github/workflows/run-tests.yml

```diff
@@ -100,6 +105,41 @@ jobs:
         run: |
           cat "$GITHUB_OUTPUT"
 
+      - name: Checkout Code
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          ref: ${{ env.COMMIT }}
+          fetch-depth: 0
+
+      - name: Fetch base branch
+        run: git fetch origin ${{ github.event.pull_request.base.ref }}:${{ github.event.pull_request.base.ref }}
+
+      - name: Compute merge base
+        run: |
+          MERGE_BASE="$(git merge-base ${{ env.COMMIT }} ${{ github.event.pull_request.base.ref }})"
+          echo "MERGE_BASE=$MERGE_BASE" >> "$GITHUB_ENV"
+
+      - name: Collect Modified Tests
+        id: collect_modified_tests
+        run: |
+          set -e
+          set -o pipefail
+          @echo "MERGE_BASE=$(MERGE_BASE) COMMIT=$(COMMIT)"
+          
+          go build -o ./cmd/tools/test/find_altered_tests ./cmd/tools/test/find_altered_tests.go
+          
+          OUTPUT="$(make collect-modified-tests "MERGE_BASE=${MERGE_BASE}" "COMMIT=${COMMIT}")"
```

Comment:
why is this going through the makefile at all?
why not something like
```
OUTPUT=$(go run ./cmd/tools/test/find_altered_tests.go ...)
```
?

also, why not have it generate output in the format that we can append directly to GITHUB_ENV, so then we can just do:
```
go run ./cmd/tools/test/find_altered_tests.go ... >> $GITHUB_ENV
```
?

if you're worried about unexpected variables getting set, then:
```
go run ./cmd/tools/test/find_altered_tests.go ... |
  grep -E '^(modified_unit_test_suits|modified_integration_test_suites|...)=' >> $GITHUB_ENV
```

Reply:
It looked like the convention was to delegate to a Make target. Happy to change this

Reply:
This seems like kind of a special "only in CI" thing. Makefile targets can be for things that people might at least in theory run manually? If you want a makefile target, I would want it do the whole thing, i.e. collect modified suits and run them x times, so people can run that themselves before creating the PR